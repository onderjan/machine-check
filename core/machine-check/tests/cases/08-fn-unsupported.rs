#[machine_check_macros::machine_description]
mod machine {
    struct A {}

    impl A {
        default fn with_defaultness() -> u32 {
            0
        }
        fn without_return_statement() -> u32 {}

        const fn with_constness() -> u32 {}
        async fn with_asyncness() -> u32 {}
        unsafe fn with_unsafety() -> u32 {}
        extern "C" fn with_abi() -> u32 {}
        fn with_variadic_arg(...) -> u32 {}

        // TODO: enable after these are no longer rejected by visitor
        //fn with_receiver_lifetime(&'a self) -> u32 {}
        //fn with_mutable_receiver(&mut self) -> u32 {}
        //fn with_mutable_arg(a: &mut u32) -> u32 {}
    }
}

fn main() {}
