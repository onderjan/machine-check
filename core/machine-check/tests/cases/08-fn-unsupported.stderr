error: machine-check: Defaultness not supported
 --> tests/cases/08-fn-unsupported.rs:6:9
  |
6 |         default fn with_defaultness() -> u32 {
  |         ^^^^^^^

error: machine-check: Functions without return statement not supported
 --> tests/cases/08-fn-unsupported.rs:9:9
  |
9 |         fn without_return_statement() -> u32 {}
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: machine-check: Constness not supported
  --> tests/cases/08-fn-unsupported.rs:11:9
   |
11 |         const fn with_constness() -> u32 {}
   |         ^^^^^

error: machine-check: Asyncness not supported
  --> tests/cases/08-fn-unsupported.rs:12:9
   |
12 |         async fn with_asyncness() -> u32 {}
   |         ^^^^^

error: machine-check: Unsafety not supported
  --> tests/cases/08-fn-unsupported.rs:13:9
   |
13 |         unsafe fn with_unsafety() -> u32 {}
   |         ^^^^^^

error: machine-check: ABI not supported
  --> tests/cases/08-fn-unsupported.rs:14:9
   |
14 |         extern "C" fn with_abi() -> u32 {}
   |         ^^^^^^^^^^

error: machine-check: Variadic argument not supported
  --> tests/cases/08-fn-unsupported.rs:15:30
   |
15 |         fn with_variadic_arg(...) -> u32 {}
   |                              ^^^

error: machine-check: Lifetimes not supported
  --> tests/cases/08-fn-unsupported.rs:17:36
   |
17 |         fn with_receiver_lifetime(&'a self) -> u32 {}
   |                                    ^^

error: machine-check: Mutable receiver argument not supported
  --> tests/cases/08-fn-unsupported.rs:18:35
   |
18 |         fn with_mutable_receiver(&mut self) -> u32 {}
   |                                   ^^^

error: machine-check: Mutable references not supported
  --> tests/cases/08-fn-unsupported.rs:19:33
   |
19 |         fn with_mutable_arg(a: &mut u32) -> u32 {}
   |                                 ^^^
